lib/datadog/core/crashtracking/component.rb

```diff
@@ -81,8 +84,42 @@ def stop
           logger.error("Failed to stop crash tracking: #{e.message}")
         end
 
+        def runtime_callback_registered?
+          return false if Datadog::Core::RUNTIME_STACKS_FAILURE
+          return false unless Datadog.const_defined?(:RuntimeStacks, false)
+
+          Datadog::RuntimeStacks._native_is_runtime_callback_registered
+        rescue => e
+          logger.debug("Runtime stack callback status check not available: #{e.message}")
+          false
+        end
+
         private
 
+        def register_runtime_stack_callback
+          # Check if runtime_stacks extension loaded successfully
+          if Datadog::Core::RUNTIME_STACKS_FAILURE
+            logger.debug("Skipping runtime stack callback registration: #{Datadog::Core::RUNTIME_STACKS_FAILURE}")
+            return
+          end
+
+          unless Datadog.const_defined?(:RuntimeStacks, false)
+            logger.debug('Skipping runtime stack callback registration: Datadog::RuntimeStacks not defined')
+            return
+          end
+
+          success = Datadog::RuntimeStacks._native_register_runtime_stack_callback
+
+          unless success
+            error_message = 'Failed to register runtime stack callback: registration returned false'
+            logger.error(error_message)
+            raise StandardError, error_message
+          end
+        rescue => e
+          logger.error("Failed to register runtime stack callback: #{e.message}")
+          raise
+        end
```

Comment:
This pattern is a bit weird -- if `success == false` we log an error, then turn it into an exception, then immediately log it again, at error level.

Something logged at error level will show up quite noisly in customer's logs so for something like crashtracking where if it doesn't work the impact is very low, I'm not sure "alarming" customers in this way is worth it...